### PR TITLE
feature: kill tracking

### PR DESCRIPTION
**Pull request by ArturKnopik**
image

based on https://github.com/otland/forgottenserver/pull/3757
Co-Authored-By: @SaiyansKing and @omarcopires 
- [x] I have followed [proper The Forgotten Server code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### CODE REVIEW

**Review by EPuncker — APPROVED** [review]

**Review by DSpeichert — APPROVED** [review]

File: data/lib/core/player.lua [hunk 1]
```diff
@@ -312,3 +312,44 @@ function Player.getWeaponType(self)
 	end
 	return WEAPON_NONE
 end
+
+local function prepareUpdateKillTrackerMsg(monster, corpse)
+	monsterType = monster:getType()
+	if monsterType then
+		return nil
+	end
```
Comment (Znote) [inline data/lib/core/player.lua:320]: Should monsterType be global? Did you forget to add `local`?
And the conditional statement looks weird, If monsterType is set, return nil? Did you forget to add `not`?

Comment (Zbizu) [reply data/lib/core/player.lua:320]: correct, should be local

File: data/lib/core/player.lua [hunk 2]
```diff
@@ -312,3 +312,44 @@ function Player.getWeaponType(self)
 	end
 	return WEAPON_NONE
 end
+
+local function prepareUpdateKillTrackerMsg(monster, corpse)
+	monsterType = monster:getType()
```
Comment (Zbizu) [inline data/lib/core/player.lua:317]: local

File: data/lib/core/player.lua [hunk 3]
```diff
@@ -312,3 +312,44 @@ function Player.getWeaponType(self)
...
+	for a = corpse:getSize() - 1, 0, -1 do
+        local item = corpse:getItem(a)
+		networkMessage:addItem(item)
+	end
+	return networkMessage
+end
+
+function Player.updateKillTracker(self, monster, corpse)
+	monsterType = monster.getType()
+	msg = prepareUpdateKillTrackerMsg(monster, corpse)
```
Comment (Zbizu) [inline data/lib/core/player.lua:340]: these two should also be local local

File: data/lib/core/player.lua (hunk 2 above)
Comment (Anastaciaa) [reply data/lib/core/player.lua:317]: unused?

**Pull request merged**